build_tag_push.py

```diff
@@ -1,101 +1,130 @@
-	name = dict_of_contents['name']
-	#print name
-	build_script = dict_of_contents['build']
-	#print build_script
-	test_call = dict_of_contents['test']
-	#print test_call
-	docker_context_build = dict_of_contents['docker_context'][0]
-	#print docker_context_build
-	docker_context_static = dict_of_contents['docker_context'][1]
-	#print docker_context_static
-	return name, build_script, test_call, docker_context_build, docker_context_static # Returns a list of the values from the dictionary dict_of_contents
-
-# Step 2 - Builds and test the app according to that config file = NOT COMPLETE.
-# There needs to be a Build File .travis.yml on their github
-# Seems as though the GitHub repo needs to be added to the Travis CI account and then when there is a new push it will
-# run the tests in the .travis.yml file that is in that repo. Ask for clarification...
-#
-#
-#
-#
-#
-#
-#
-#
-#
-#
-
-# Step 3 - Runs docker build with the right context = PARTIALLY COMPLETE.
-# They will have their own Dockerfile in their GitHub repo (I am assuming) and this file is a text file with 
-# instructions to build image. Dockerfile -> docker build -> Docker Image, it is basically an automation of Docker Image
-# Creation. In the DockerFile, you generally do: FROM = tells Docker which image you base your image on, 
+	# name = dict_of_contents['name']
+	# build_script = dict_of_contents['build']
+	# test_call = dict_of_contents['test']
+	# docker_context_build = dict_of_contents['docker_context'][0]
+	# docker_context_static = dict_of_contents['docker_context'][1]
+	return dict_of_contents
+
+# Step 2 - Builds and test the app according to that config file = PENDING.
+
+# Is this something other than working with Travis CI or is this part of the next step? 
+# I was under the impression that Travis CI is what we are using to build and test their app. 
+# If not, can you elaborate more on what you want this step to be?
+
+def createContextDirectory(contextArray):
+	original_dir_path = os.getcwd() # Getting this path to copy over files later in this function. I am assuming
+									# this is where the original files were put that we will be working out of.
+
+	dir_name = "MVPStudio_temp_dir" 
+	# Should this be user input so it can be different names per group? -->
+	# dir_name = input('Enter a directory name for your Docker Context: ')
+
+	os.makedirs(dir_name, exist_ok=True) # Making the directory and saying to not raise an error if the directory already exists.
+
+	dir_path = original_dir_path+'/'+dir_name # This is providing the path to the Docker Context directory.
+
+	copyfile(original_dir_path+'/'+contextArray[0], dir_path+'/'+contextArray[0]) # Copying the first file from docker_context
+																				  # and putting it in the Docker Context directory 
+																				  # created.
+	copyfile(original_dir_path+'/'+contextArray[1], dir_path+'/'+contextArray[1]) # Copying the first file from docker_context
```

Comment:
By doing an explicit `[0]` and `[1]` your script works only if there's *exactly* two items in their context. We want to support any number of items in the array so do a loop. Also, `copyfile` will only copy a single file but in most cases a build has **hundreds** of outputs so most items in the context array will be directories that need to be copied recusively. I think you need something like (untested):

```
for context_item in contextArray:
   if os.path.isdir(context_item):
        copytree(os.path.join(orginal_dir_path, context_item)), os.path.join(context_base, context_item))
   else:
        copyfile(....)
```